components/suggest/src/store.rs

```diff
@@ -348,31 +323,76 @@ where
                     })?;
                 }
                 SuggestRecord::Amo => {
-                    let Some(attachment) = record.attachment.as_ref() else {
-                        writer.write(|dao| dao.put_last_ingest_if_newer(record.last_modified))?;
-                        continue;
-                    };
-
-                    let attachment: SuggestAttachment<_> = serde_json::from_slice(
-                        &self.settings_client.get_attachment(&attachment.location)?,
+                    self.ingest_suggestions_from_record(
+                        writer,
+                        record,
+                        |dao, record_id, suggestions| {
+                            dao.insert_amo_suggestions(record_id, suggestions)
+                        },
+                    )?;
+                }
+                SuggestRecord::Pocket => {
+                    self.ingest_suggestions_from_record(
+                        writer,
+                        record,
+                        |dao, record_id, suggestions| {
+                            dao.insert_pocket_suggestions(record_id, suggestions)
+                        },
                     )?;
-
-                    writer.write(|dao| {
-                        dao.drop_suggestions(&record_id)?;
-
-                        dao.insert_amo_suggestions(&record_id, attachment.suggestions())?;
-
-                        dao.drop_unparsable_record_id(&record_id)?;
-
-                        dao.put_last_ingest_if_newer(record.last_modified)?;
-
-                        Ok(())
-                    })?;
                 }
             }
         }
         Ok(())
     }
+
+    fn ingest_suggestions_from_record<T>(
+        &self,
+        writer: &SuggestDb,
+        record: &RemoteSettingsRecord,
+        block: impl FnOnce(&mut SuggestDao<'_>, &SuggestRecordId, &[T]) -> Result<()>,
```

Comment:
Nit: let's call it `ingestion_handler` or the like as `block` has special meanings in the context of database ops.